src/pk/ecc/ecc_sign_hash.c

```diff
@@ -69,6 +84,16 @@ static int _ecc_sign_hash(const unsigned char *in,  unsigned long inlen,
       /* find r = x1 mod n */
       if ((err = mp_mod(pubkey.pubkey.x, p, r)) != CRYPT_OK)               { goto error; }
 
+      /* find recovery ID */
```

Comment:
It's an independent property of the signature - the `recid` output isn't needed for LTC_ECCSIG_ETH27, since that contains the value, but can be used to recover keys with a signature in a different format if separately transmitted.

I could put the calculation in an `if (recid || sigformat==LTC_ECCSIG_ETH27)` block so it's only calculated if it's going to be used?